src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Logging/ExportLoggerProviderAttribute.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Razor.Logging;
+
+namespace Microsoft.VisualStudio.Razor.Logging;
+
+[MetadataAttribute]
+[AttributeUsage(AttributeTargets.Class)]
+internal sealed class ExportLoggerProviderAttribute : ExportAttribute
+{
+    public LogLevel? MinimumLogLevel { get; }
+
+    public ExportLoggerProviderAttribute()
+        : base(typeof(ILoggerProvider))
+    {
+        MinimumLogLevel = null;
+    }
+
+    public ExportLoggerProviderAttribute(LogLevel minimumLogLevel)
```

Comment:
This might be MEF magic I'm not aware of, won't we still need to load the image to check what the min level is?

Reply:
oh... is that the logger provider won't initialize until a min level is met? And that prevents IT from loading dependencies? slick

Reply:
That's right. It's really only viable for one of the logger providers (the activity log provider), since the others either track the log level option or just handle all log levels, but it's something.